File: yfwf-1-py-process-numbers-list-user/task.py

import argparse
import json
import os
arg_parser = argparse.ArgumentParser()


arg_parser.add_argument('--id', action='store', type=str, required=True, dest='id')


arg_parser.add_argument('--numbers', action='store', type=str, required=True, dest='numbers')


args = arg_parser.parse_args()
print(args)

id = args.id

numbers = json.loads(args.numbers)



new_numbers = []
for number in numbers:
    new_numbers.append(number + 1 )

file_new_numbers = open("/tmp/new_numbers_" + id + ".json", "w")
file_new_numbers.write(json.dumps(new_numbers))
file_new_numbers.close()

File: r-serialize-to-json-string-user/task.R
setwd('/app')
library(optparse)
library(jsonlite)

if (!requireNamespace("jsonlite", quietly = TRUE)) {
	install.packages("jsonlite", repos="http://cran.us.r-project.org")
}
library(jsonlite)



print('option_list')
option_list = list(

make_option(c("--id"), action="store", default=NA, type="character", help="task id")
)


opt = parse_args(OptionParser(option_list=option_list))

var_serialization <- function(var){
    if (is.null(var)){
        print("Variable is null")
        exit(1)
    }
    tryCatch(
        {
            var <- fromJSON(var)
            print("Variable deserialized")
            return(var)
        },
        error=function(e) {
            print("Error while deserializing the variable")
            print(var)
            var <- gsub("'", '"', var)
            var <- fromJSON(var)
            print("Variable deserialized")
            return(var)
        },
        warning=function(w) {
            print("Warning while deserializing the variable")
            var <- gsub("'", '"', var)
            var <- fromJSON(var)
            print("Variable deserialized")
            return(var)
        }
    )
}

id <- gsub('"', '', opt$id)


print("Running the cell")
data <- list(
  task = "Write Python",
  number = 1,
  skills = c("python", "sql")
)

json_str <- toJSON(data, pretty = TRUE, auto_unbox = TRUE)

print(json_str)
# capturing outputs
print('Serialization of data')
file <- file(paste0('/tmp/data_', id, '.json'))
writeLines(toJSON(data, auto_unbox=TRUE), file)
close(file)
print('Serialization of json_str')
file <- file(paste0('/tmp/json_str_', id, '.json'))
writeLines(toJSON(json_str, auto_unbox=TRUE), file)
close(file)
